ORBIT/core/logic/vessel_logic.py

```diff
@@ -50,6 +48,71 @@ def prep_for_site_operations(vessel, survey_required=False, **kwargs):
         )
 
 
+@process
```

Comment:
New generic stabilization routine. If the vessel has DP, it returns without taking any time. Otherwise it checks for jacking system and uses the previous methods.